### PR TITLE
general: Port to GTK4+libadwaita

### PR DESCRIPTION
**Pull request by BrainBlasted**
* Ports to GTK4+libadwaita
* Use libadwaita styling where wanted
* Rework menus to use new APIs

### CODE REVIEW

**Comment by BrainBlasted** [comment]
I wouldn't recommend merging this until libadwaita 1.0 is out.

**Comment by rafaelmardojai** [comment]
Thanks for making progress on this! Really appreciated. 

> I wouldn't recommend merging this until libadwaita 1.0 is out.

Yep, my idea was to start porting after libadwaita 1.0 came out.

File: data/resources/preferences.ui [hunk 1]
```diff
@@ -2,28 +2,22 @@
 <interface>
   <requires lib="gtk+" version="3.0"/>
   <requires lib="libhandy" version="1.0"/>
```
Comment (A6GibKm) [inline data/resources/preferences.ui:4]: These lines should be removed.

File: data/resources/style.css [hunk 2]
```diff
@@ -1,19 +1,7 @@
...
-  font-weight: bold;
-}
-
-.h4 {
-  font-weight: bold;
-}
-
 .soundrow.playing .sound-icon {
-  color: @theme_selected_bg_color;
+  color: @accent_color;
```
Comment (A6GibKm) [inline data/resources/style.css:2]: Now there are two variables: accent_bg_color, accent_fg_color. accent_color was removed.

File: data/resources/window.ui [hunk 3]
```diff
@@ -2,68 +2,42 @@
 <!-- Generated with glade 3.22.0 -->
 <interface>
   <requires lib="gtk+" version="3.24"/>
-  <template class="BlanketWindow" parent="HdyApplicationWindow">
-    <property name="can_focus">False</property>
+  <template class="BlanketWindow" parent="AdwApplicationWindow">
     <property name="default_width">600</property>
     <property name="default_height">600</property>
+    <property name="title" translatable="yes">Blanket</property>
```
Comment (A6GibKm) [inline data/resources/window.ui:8]: Is this necessary?, setting g_set_application_name should automatically se it as title for windows and about dialog.

File: data/resources/sound-row.ui [hunk 4]
```diff
@@ -59,19 +50,12 @@
             </child>
             <child>
               <object class="GtkScale" id="volume">
-                <property name="visible">True</property>
-                <property name="can_focus">True</property>
+                <property name="hexpand">True</property>
                 <property name="adjustment">volume_adjustment</property>
```
Comment (A6GibKm) [inline data/resources/sound-row.ui:54]: Not really an issue, but you can directly define the GtkAdjustment here (like the child property).

File: blanket/main.py [hunk 5]
```diff
@@ -24,7 +24,7 @@
 from blanket.about import AboutDialog
 
 
-class Application(Gtk.Application):
+class Application(Adw.Application):
     def __init__(self, version):
         super().__init__(application_id='com.rafaelmardojai.Blanket',
                          flags=Gio.ApplicationFlags.HANDLES_COMMAND_LINE)
```
Comment (A6GibKm) [inline blanket/main.py:30]: In the line bellow this one, 

```
        GLib.set_prgname('com.rafaelmardojai.Blanket')
```

should be removed.

**Review by A6GibKm — COMMENTED** [review]
Some minor nits.

File: data/resources/style.css (hunk 2 above)
Comment (BrainBlasted) [reply data/resources/style.css:2]: No - `@accent_color` is still there, and it's meant to be used for icons or text. https://gitlab.gnome.org/GNOME/libadwaita/-/blob/main/src/stylesheet/_colors.scss#L6

File: blanket/main.py (hunk 5 above)
Comment (BrainBlasted) [reply blanket/main.py:30]: Why should it be removed?

Comment (A6GibKm) [reply blanket/main.py:30]: From the migration guide https://docs.gtk.org/gtk4/migrating-3to4.html#set-a-proper-application-id

> The call to g_set_prgname() can be removed once you fully migrated to GTK 4.

It seems to be redundant.

**Comment by TheOPtimal** [comment]
In `preferences.py`, line 38 gtk prefer dark is used, but AdwStyleManager should be used there instead

**Comment by rafaelmardojai** [comment]
The branch has conflicts :p

**Pull request merged**